Controller upgrade to v1.5.0 fails to deploy new BIG-IP configuration

#### Description
When upgrading from a previous version of the controller to v1.5.0, the BIG-IP configurations fail to deploy.

#### OpenShift Version
3.7.1

#### Controller Version
1.5.0

#### BIG-IP Version
<Version of BIG-IP being used>

#### Diagnostic Information
The controller logs show the following messages soon after startup:
```
2018/04/25 14:57:16 [ERROR] [2018-04-25 14:57:16,113 __main__ ERROR] Unexpected error
2018/04/25 14:57:16 [INFO] Traceback (most recent call last):
2018/04/25 14:57:16 [INFO]   File "/app/src/f5-ctlr-agent/f5_ctlr_agent/bigipconfigdriver.py", line 323, in _do_reset
2018/04/25 14:57:16 [INFO]     incomplete = self._update_cccl(config)
2018/04/25 14:57:16 [INFO]   File "/app/src/f5-ctlr-agent/f5_ctlr_agent/bigipconfigdriver.py", line 397, in _update_cccl
2018/04/25 14:57:16 [INFO]     incomplete += mgr._apply_ltm_config(cfg_ltm)
2018/04/25 14:57:16 [INFO]   File "/app/src/f5-ctlr-agent/f5_ctlr_agent/bigipconfigdriver.py", line 117, in _apply_ltm_config
2018/04/25 14:57:16 [INFO]     return self._cccl.apply_ltm_config(config)
2018/04/25 14:57:16 [INFO]   File "/usr/local/lib/python2.7/site-packages/f5_cccl/api.py", line 92, in apply_ltm_config
2018/04/25 14:57:16 [INFO]     self._user_agent)
2018/04/25 14:57:16 [INFO]   File "/usr/local/lib/python2.7/site-packages/f5_cccl/service/manager.py", line 670, in apply_ltm_config
2018/04/25 14:57:16 [INFO]     desired_config, default_route_domain)
2018/04/25 14:57:16 [INFO]   File "/usr/local/lib/python2.7/site-packages/f5_cccl/service/manager.py", line 392, in deploy_ltm
2018/04/25 14:57:16 [INFO]     self._get_resource_tasks(existing_virtuals, desired)[0:4])
2018/04/25 14:57:16 [INFO]   File "/usr/local/lib/python2.7/site-packages/f5_cccl/service/manager.py", line 81, in _get_resource_tasks
2018/04/25 14:57:16 [INFO]     if desired[resource] != managed[resource]
2018/04/25 14:57:16 [INFO]   File "/usr/local/lib/python2.7/site-packages/f5_cccl/resource/resource.py", line 112, in __ne__
2018/04/25 14:57:16 [INFO]     return not self.__eq__(resource)
2018/04/25 14:57:16 [INFO]   File "/usr/local/lib/python2.7/site-packages/f5_cccl/resource/ltm/virtual.py", line 191, in __eq__
2018/04/25 14:57:16 [INFO]     len(self._data[key]) != len(other.data.get(key, None)):
2018/04/25 14:57:16 [INFO] TypeError: object of type 'NoneType' has no len()
2018/04/25 14:57:16 [ERROR] [2018-04-25 14:57:16,117 __main__ ERROR] Error applying config, will try again in 1 seconds
 ```

--- a/cmd/k8s-bigip-ctlr/main.go
+++ b/cmd/k8s-bigip-ctlr/main.go
@@ -375,12 +375,11 @@ func setupNodePolling(
 	eventChan <-chan interface{},
 	kubeClient kubernetes.Interface,
 ) error {
-	if appMgr.IsNodePort() {
-		err := np.RegisterListener(appMgr.ProcessNodeUpdate)
-		if nil != err {
-			return fmt.Errorf("error registering node update listener for nodeport mode: %v",
-				err)
-		}
+	// Register appMgr to watch for node updates to keep track of watched nodes
+	err := np.RegisterListener(appMgr.ProcessNodeUpdate)
+	if nil != err {
+		return fmt.Errorf("error registering node update listener: %v",
+			err)
 	}
 
 	if 0 != len(vxlanMode) {
@@ -406,12 +405,6 @@ func setupNodePolling(
 		err = np.RegisterListener(vxMgr.ProcessNodeUpdate)
 		if nil != err {
 			return fmt.Errorf("error registering node update listener for vxlan mode: %v",
-				err)
-		}
-		// Register appMgr to watch for node updates to keep track of watched nodes
-		err = np.RegisterListener(appMgr.ProcessNodeUpdate)
-		if nil != err {
-			return fmt.Errorf("error registering node update listener for appManager: %v",
 				err)
 		}
 		if eventChan != nil {
@@ -599,7 +592,8 @@ func main() {
 		log.Fatalf("error connecting to the client: %v", err)
 	}
 	if *manageRoutes {
-		rclient, err := routeclient.New(config)
+		var rclient *routeclient.Client
+		rclient, err = routeclient.New(config)
 		appMgrParms.RouteClientV1 = rclient.RESTClient
 		if nil != err {
 			log.Fatalf("unable to create route client: err: %+v\n", err)
@@ -608,18 +602,16 @@ func main() {
 
 	appMgr := appmanager.NewManager(&appMgrParms)
 
-	if isNodePort || 0 != len(vxlanMode) {
-		intervalFactor := time.Duration(*nodePollInterval)
-		np := pollers.NewNodePoller(appMgrParms.KubeClient, intervalFactor*time.Second, *nodeLabelSelector)
-		err := setupNodePolling(appMgr, np, eventChan, appMgrParms.KubeClient)
-		if nil != err {
-			log.Fatalf("Required polling utility for node updates failed setup: %v",
-				err)
-		}
-
-		np.Run()
-		defer np.Stop()
+	intervalFactor := time.Duration(*nodePollInterval)
+	np := pollers.NewNodePoller(appMgrParms.KubeClient, intervalFactor*time.Second, *nodeLabelSelector)
+	err = setupNodePolling(appMgr, np, eventChan, appMgrParms.KubeClient)
+	if nil != err {
+		log.Fatalf("Required polling utility for node updates failed setup: %v",
+			err)
 	}
+
+	np.Run()
+	defer np.Stop()
 
 	setupWatchers(appMgr, 30*time.Second)
 	// Expose Prometheus metrics

--- a/docs/RELEASE-NOTES.rst
+++ b/docs/RELEASE-NOTES.rst
@@ -4,8 +4,13 @@ Release Notes for BIG-IP Controller for Kubernetes
 next-release
 ------------
 
+v1.5.1
+------
+
 Bug Fixes
 `````````
+* :issues:`683` - Controller upgrades properly with new metadata field.
+* :issues:`686` - Controller in cluster mode does not rely on vxlan name to configure pool members.
 
 v1.5.0
 ------
@@ -40,6 +45,9 @@ Bug Fixes
 Limitations
 ```````````
 * Cannot apply app-root and url-rewrite annotations to the same resource; see: :issues:`675`
+* If an older controller created resources, upgrading to the new version could
+  result in a python exception when adding metadata to virtuals: :issues:`683`
+* If running the controller in cluster mode without a vxlan name, pool members are not created: :issues:`686`
 
 v1.4.2
 ------

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,1 +1,1 @@
--e git+https://github.com/f5devcentral/f5-ctlr-agent.git@ffe7c22288502ca45f14bcf2b6d93295ca58a593#egg=f5-ctlr-agent
+-e git+https://github.com/f5devcentral/f5-ctlr-agent.git@435f049bf1a36c56293e57686d88a0bcd03e4cf6#egg=f5-ctlr-agent

